docs/pipeline-concept.md

```diff
@@ -0,0 +1,208 @@
+3. `variables`: A list of environment variables that are set before executing the script.
+4. `variables.name`: The name of the environment variable.
+5. `variables.configRef`: The [configuration reference](configuration.md) to look up the value for the environment variable.
+6. `variables.value`: Alternatively static values can be provided for the environment variable.
+
+Currently, the following list of tools can be used within shell scripts:
+
+- `az`
+- `helm`
+- `kubectl`
+- `jq`
+- `make`
+- ...
+
+See the [Shell extension](https://ev2docs.azure.net/features/service-artifacts/actions/shell-extensions/overview.html) documentation for more details.
+
+Shell steps also support dry-run testing, but such scripts need to explicitely implement it and mark support for it with the `dryRun` property.
+
+```yaml
+  ...
+  steps:
+  - ...
+    action: Shell
+    ...
+    dryRun:                                          (1)
+      variables:
+      - name: DRY_RUN                                (2)
+        value: "true"
+```
+
+1. `dryRun`: Marks the step as supporting dry-run testing.
+2. `variables`: A list of environment variables that are set before executing the script.
+
+It is the scripts responsibility to react to the `DRY_RUN` environment variable correctly and not perform any real update actions on the target subscription/resourcegroup/AKS cluster.
+
+>[!WARNING]
+> TODO: we need to align and document the tool versions between the EV2 execution context and the Red Hat pipeline runner.
+
+Shell steps are mostly used for service deployments leveraging [Helm charts](service-deployment-concept.md#helm-chart).
+
+### Step execution context
+
+All steps share a common execution context:
+
+#### Azure session
+
+When executing step, an Azure session is provided for the defined subscription (`resourceGroups.subscription`) and resourcegroup (`resourceGroups.name`), allowing authenticated Azure operations. The identity used for this session is provided in the [configuration](configuration.md) as `aroDevopsMsiId`.
+
+> [!IMPORTANT]
+> Please note that this identity does not have access to all Azure resources in our subscriptions and resourcegroups by default. The necessary permissions need to be granted explicitly. You can observe this in various Bicep templates, where this identity is granted specific permissions, e.g. on Key Vaults or storage accounts.
+
+#### Azure resource groups
+
+The resourcegroup (`resourceGroups.name`) is pre-created before step execution starts.
+
+#### AKS cluster
+
+If an `resourceGroups.aksCluster` is specified, the `KUBECONFIG` environment variable is set and allows cluster admin interaction withe the AKS cluster. This is mostly relevant for `Shell` steps.
+
+## Pipeline Deployment Scope
+
+A pipeline is the smallest unit of deployment in ARO HCP. This means that all steps within a pipeline are executed from start to finish—there is no concept of executing a single step in only.
```

Comment:
Can you please outline the difference to usage of the `scope` context in Azure deployments?

Reply:
i added a section that specifically outlines the difference between the architecturla scope and the ARM step `deploymentLevel`